clouddriver-cloudfoundry/src/main/java/com/netflix/spinnaker/clouddriver/cloudfoundry/client/CloudFoundryClientUtils.java

```diff
@@ -55,7 +55,7 @@ static void safelyCall(RetrofitConsumer r) throws CloudFoundryApiException {
       .orElseThrow(() -> new CloudFoundryApiException("Unable to retrieve " + resourceNamePluralized));
 
     List<R> allResources = new ArrayList<>(firstPage.getResources());
-    for (int page = 2; page < Math.max(2, firstPage.getPagination().getTotalPages()); page++) {
+    for (int page = 2; page <= Math.max(2, firstPage.getPagination().getTotalPages()); page++) {
       final int p = page;
       allResources.addAll(safelyCall(() -> fetchPage.apply(p))
         .orElseThrow(() -> new CloudFoundryApiException("Unable to retrieve " + resourceNamePluralized))
```

Comment:
I see very similar logic on line 73 below.
Does it suffer from the same error and require a "<=" too?
If so, then isn't there a way to avoid repeating the logic so that there aren't multiple places to update in the first place?

Reply:
I've added the logic and an additional test for line 73.

you are correct that we could try to refactor that logic to avoid repeating the code. 
I'm going to look at that and do a separate PR to remove the duplicate logic next. I think it should be a separate PR because a change would require changes in our model and how `page` and `pagination` object are handled.